server/src/main/java/com/objectcomputing/checkins/services/memberSkill/MemberSkillServiceImpl.java

```diff
@@ -0,0 +1,93 @@
+package com.objectcomputing.checkins.services.memberSkill;
+
+import com.objectcomputing.checkins.services.memberprofile.MemberProfileRepository;
+import com.objectcomputing.checkins.services.skills.SkillRepository;
+
+import javax.inject.Inject;
+import javax.validation.constraints.NotNull;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.UUID;
+
+public class MemberSkillServiceImpl implements MemberSkillServices {
+
+    @Inject
+    private MemberSkillRepository memberSkillRepository;
+
+    @Inject
+    private MemberProfileRepository memberProfileRepository;
+
+    @Inject
+    private SkillRepository skillRepository;
+
+    public MemberSkill save(MemberSkill memberSkill) {
+        MemberSkill memberSkillRet = null;
+        if (memberSkill != null) {
+            final UUID memberId = memberSkill.getMemberid();
+            final UUID skillId = memberSkill.getSkillid();
+            if (skillId == null || memberId == null) {
+                throw new MemberSkillBadArgException(String.format("Invalid member skill %s", memberSkill));
+            } else if (memberSkill.getId() != null) {
+                throw new MemberSkillBadArgException(String.format("Found unexpected id %s for member skill", memberSkill.getId()));
+            } else if (!memberProfileRepository.findById(memberId).isPresent()) {
+                throw new MemberSkillBadArgException(String.format("Member Profile %s doesn't exist", memberId));
+            } else if (!skillRepository.findById(skillId).isPresent()) {
+                throw new MemberSkillBadArgException(String.format("Skill %s doesn't exist", skillId));
+            } else if (memberSkillRepository.findByMemberidAndSkillid(memberSkill.getMemberid(),
+                    memberSkill.getSkillid()).isPresent()) {
+                throw new MemberSkillBadArgException(String.format("Member %s already has this skill %s", memberId, skillId));
+            }
+
+            memberSkillRet = memberSkillRepository.save(memberSkill);
+        }
+        return memberSkillRet;
+
+    }
+
+    public Set<MemberSkill> readAll() {
+        Set<MemberSkill> memberSkillList = new HashSet<>();
+        memberSkillRepository.findAll().forEach(memberSkillList::add);
+        return memberSkillList;
+    }
```

Comment:
You don't need this...because of your `findByfields`

Reply:
👍